API/Controllers/AdminNotesController.cs

```diff
@@ -0,0 +1,169 @@
+            entityId,
+            User.GetSubjectId(),
+            note
+        );
+
+        return result is not null
+            ? CreatedAtAction(
+                "ListNotes",
+                new { entity = DynamicRouteTarget.Original, entityId },
+                result
+            )
+            : BadRequest();
+    }
+
+    /// <summary>
+    /// List admin notes for an entity
+    /// </summary>
+    /// <param name="entityId">Entity id</param>
+    /// <response code="404">An entity matching the given id does not exist</response>
+    /// <response code="200">Returns all admin notes for the entity</response>
+    [HttpGet("{entityId:int}/notes")]
+    [Authorize(Roles = $"{OtrClaims.Roles.User}, {OtrClaims.Roles.Client}")]
+    [ProducesResponseType(StatusCodes.Status404NotFound)]
+    [ProducesResponseType<IEnumerable<AdminNoteDTO>>(StatusCodes.Status200OK)]
+    public async Task<IActionResult> ListNotesAsync(int entityId)
+    {
+        // Check if target entity exists
+        if (await context.FindAsync(DynamicRouteTarget.EntityType, entityId) is null)
+        {
+            return NotFound();
+        }
+
+        IEnumerable<AdminNoteDTO> result = await DynamicAdminNoteService.ListAsync(entityId);
+
+        return Ok(result);
+    }
+
+    /// <summary>
+    /// Update an admin note
+    /// </summary>
+    /// <param name="noteId">Admin note id</param>
+    /// <param name="note">New content of the admin note</param>
+    /// <response code="404">An admin note matching the given noteId does not exist </response>
+    /// <response code="400">The update was not successful</response>
+    /// <response code="200">Returns the updated admin note</response>
+    [HttpPatch("notes/{noteId:int}")]
+    [Authorize(Roles = OtrClaims.Roles.Admin)]
+    [ProducesResponseType(StatusCodes.Status404NotFound)]
+    [ProducesResponseType(StatusCodes.Status400BadRequest)]
+    [ProducesResponseType<AdminNoteDTO>(StatusCodes.Status201Created)]
+    public async Task<IActionResult> UpdateNoteAsync(
+        int noteId,
+        [FromBody][Required] string note
+    )
+    {
+        AdminNoteDTO? existingNote = await DynamicAdminNoteService.GetAsync(noteId);
+
+        if (existingNote is null)
+        {
+            return NotFound();
+        }
+
+        existingNote.Note = note;
+        AdminNoteDTO? updatedNote = await DynamicAdminNoteService.UpdateAsync(existingNote);
+
+        return updatedNote is not null ? Ok(updatedNote) : BadRequest();
+    }
+
+    /// <summary>
+    /// Delete an admin note
+    /// </summary>
+    /// <param name="noteId">Admin note id</param>
+    /// <response code="404">An admin note matching the given noteId does not exist </response>
+    /// <response code="400">The deletion was not successful</response>
+    /// <response code="204">The admin note was deleted</response>
+    // [Authorize(Roles = OtrClaims.Roles.Admin)]
+    [AllowAnonymous]
```

Comment:
```suggestion
    [Authorize(Roles = OtrClaims.Roles.Admin)]
```
Missed this from testing :P